docs/plugins/reindex.asciidoc

```diff
@@ -0,0 +1,608 @@
+
+The node must be stopped before removing the plugin.
+
+[[update-by-query-usage]]
+==== Using `_update_by_query`
+
+The simplest usage of `_update_by_query` just `touch`es every document in an
+index:
+
+[source,js]
+--------------------------------------------------
+POST /twitter/_update_by_query
+--------------------------------------------------
+// AUTOSENSE
+
+That will return something like this:
+
+[source,js]
+--------------------------------------------------
+{
+  "took" : 639,
+  "updated": 1235,
+  "batches": 13,
+  "version_conflicts": 2,
+  "failures" : [ ]
+}
+--------------------------------------------------
+
+`_update_by_query` gets a snapshot of the index when it starts and indexes what
+it finds using `internal` versioning. That means that if the document changes
+between the time when the snapshot was taken and when the index request is
+processed then you'll get a version conflict. When the versions match the
+document is changed and the version number is incremented.
+
+Version conflicts are just counted and not returned by default. To count them
+you must set `conflicts=abort` on the url or `"conflicts": "abort"` in the
+request body. If you do then the `failures` element will contain version
+conflict information. There is no `conflicts=record` because we number of
+conflicts could grow without bounds so its either `abort` or nothing.
+
+Speaking of the `failures` element, it will contain a list of failures if
+there are any. The only "optional" sort of failures are version conflicts. All
+other failures always cause the operation to abort. While the first failure
+causes the abort, it's possible to encounter lots of subsequent failures due to
+batching so all failures are reported. Just in case you want them.
+
+It's important to remember that "abort" here doesn't mean "rollback". This
+operation isn't atomic. If it fails part way through then its changes have been
+made.
+
+Back to the API format, you can limit the copy to a single type. This will only
+update `tweet`s from the `twitter` index:
+
+[source,js]
+--------------------------------------------------
+POST /twitter/tweet/_update_by_query
+--------------------------------------------------
+// AUTOSENSE
+
+You can limit using the {ref}/query-dsl.html[Query DSL]. This will update
+all documents from the `twitter` index for the user `kimchy`:
+
+[source,js]
+--------------------------------------------------
+POST /twitter/_update_by_query
+{
+  "query": { <1>
+    "term": {
+      "user": "kimchy"
+    }
+  }
+}
+--------------------------------------------------
+// AUTOSENSE
+
+<1> The query must be passed as a value to the `query` key, in the same
+way as the {ref}/search-search.html[Search API]. You can also use the `q`
+parameter in the same way as the search api.
+
+So far we've only been touching documents. This is useful for picking up new
+properties if the mapping has `"dynamic": "false"` but its only half the fun.
```

Comment:
Right. I could pull the whole example up to the top but then I think it'd
be pretty distracting for folks trying to use the doc as a syntax
reference. I'm not really sure what to do!

On Wed, Jan 6, 2016 at 9:48 AM, Yannick Welsch <notifications@github.com>
wrote:

> In docs/plugins/reindex.asciidoc
> <https://github.com/elastic/elasticsearch/pull/15125#discussion_r48964055>
> :
>
> > +{
> > +  "query": { <1>
> > +    "term": {
> > +      "user": "kimchy"
> > +    }
> > +  }
> > +}
> > +--------------------------------------------------
> > +// AUTOSENSE
> > +
> > +<1> The query must be passed as a value to the `query` key, in the same
> > +way as the {ref}/search-search.html[Search API]. You can also use the `q`
> > +parameter in the same way as the search api.
> > +
> > +So far we've only been touching documents. This is useful for picking up new
> > +properties if the mapping has `"dynamic": "false"` but its only half the fun.
>
> This relates to my comment I made above about "touch". I think if we
> motivate "touching" before we head off the section, an explanation here
> becomes superfluous. What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/15125/files#r48964055>.
>